app/models/service_template.rb

```diff
@@ -290,4 +305,37 @@ def validate_order
   def provision_action
     resource_actions.find_by(:action => "Provision")
   end
+
+  def create_resource_actions(ae_endpoints)
+    ae_endpoints ||= {}
+    [
+      {:name      => 'Provision',
+       :param_key => :provision,
+       :method    => 'default_provisioning_entry_point',
+       :args      => [service_type]},
+      {:name      => 'Reconfigure',
+       :param_key => :reconfigure,
+       :method    => 'default_reconfiguration_entry_point',
+       :args      => []},
+      {:name      => 'Retirement',
+       :param_key => :retirement,
+       :method    => 'default_retirement_entry_point',
+       :args      => []}
+    ].each do |action|
+      ae_endpoint = ae_endpoints[action[:param_key]]
+      next unless ae_endpoint
+      fqname = if ae_endpoint.empty?
+                 self.class.send(action[:method], *action[:args]) || ""
+               else
+                 ae_endpoint[:fqname]
+               end
+
+      build_options = {:action => action[:name], :fqname => fqname}
+      build_options.merge!(ae_endpoint.slice(:dialog, :dialog_id))
+      build_options.merge!(ae_endpoint.slice(:configuration_template, :configuration_template_id, :configuration_template_type))
```

Comment:
Yes, we can. I split into two lines because the list is too long. Maybe we could create an array in one line and pass it to the `slice` method in the second line. @jntullo